src/main/java/com/cloudant/client/api/DesignDocumentManager.java

```diff
@@ -0,0 +1,225 @@
+ * <pre>
+ * // read from javascript design document file
+ * {@code
+ * DesignDocument design = DesignDocumentManager.fromFile("../example.js");
+ * }
+ *
+ * // read from directory of design documents
+ * {@code
+ * List<DesignDocument>DesignDocument design =
+ *                  DesignDocumentManager.fromDirectory("../design-files/");
+ * }
+ *
+ * // sync with the database
+ * {@code
+ * db.getDesignDocumentManager().put(designDoc);
+ * }
+ *
+ * // sync multiple design documents with the database
+ * {@code
+ * db.getDesignDocumentManager().put(listDesignDocuments);
+ * }
+ *
+ * // read from the database
+ * {@code
+ * DesignDocument designDoc2 = db.getDesignDocumentManager().get("_design/example");
+ * }
+ * </pre>
+ *
+ * @see {@link Database#getDesignDocumentManager()} to access the API.
+ * @see DesignDocument
+ * @since 2.0.0
+ */
+public class DesignDocumentManager {
+
+    private URI dbUri;
+    private CouchDatabaseBase dbc;
+    private CloudantClient client;
+
+    DesignDocumentManager(CouchDatabase db, CloudantClient client) {
+        this.dbUri = db.getDBUri();
+        this.dbc = db;
+        this.client = client;
+    }
+
+
+    /**
+     * Synchronizes a design document to the Database.
+     * <p>This method will first try to find a document in the database with the same id
+     * as the given document, if it is not found then the given document will be saved to the
+     * database.
+     * <p>If the document was found in the database, it will be compared with the given document
+     * using
+     * {@code equals()}. If both documents are not equal, then the given document will be saved to
+     * the
+     * database and updates the existing document.
+     *
+     * @param document The design document to synchronize
+     * @return {@link Response} as a result of a document save or update, or returns {@code null}
+     * if no
+     * action was taken and the document in the database is up-to-date with the given document.
+     */
+    public Response put(DesignDocument document) {
+        CouchDbUtil.assertNotEmpty(document, "Document");
+        DesignDocument documentFromDb = null;
+        try {
+            documentFromDb = get(document.getId());
+
+        } catch (NoDocumentException e) {
+            return dbc.save(document);
+        }
+        if (!document.equals(documentFromDb)) {
+            document.setRevision(documentFromDb.getRevision());
+            return dbc.update(document);
+        }
+        return null;
+    }
+
+    public void put(DesignDocument... designDocs) {
+        for(DesignDocument designDocument : designDocs) {
+            put(designDocument);
+        }
+    }
+
+    /**
+     * Gets a design document from the database.
+     *
+     * @param id The document id
+     * @return {@link DesignDocument}
+     */
+    public DesignDocument get(String id) {
+        final URI uri = buildUri(dbUri).path(id).build();
+        return client.get(uri, DesignDocument.class);

```

Comment:
The methods in here will need to use `client.couchDbClient.get(...)` etc otherwise these will break when it merges to master because of #133